src/main/java/gr/gousiosg/javacg/stat/JCallGraph.java

```diff
@@ -92,17 +102,71 @@ public static void main(String[] args) {
           // Build and serialize a staticcallgraph object with jar files provided
           BuildArguments arguments = new BuildArguments(args);
           StaticCallgraph callgraph = StaticCallgraph.build(arguments);
+          callgraph.JarEntry=arguments.getJars().get(0).first;
           maybeSerializeStaticCallGraph(callgraph, arguments);
           break;
         }
+        case "buildyaml":{
+          DumperOptions options = new DumperOptions();
+          options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+          options.setPrettyFlow(true);
+          Yaml yaml = new Yaml(options);
+
+          JarInputStream jarFileStream = new JarInputStream(new FileInputStream(args[1]));
+          JarFile jarFile = new JarFile(args[1]);
+
+          ArrayList<JarEntry> listOfAllClasses = getAllClassesFromJar(jarFileStream);
+          ArrayList<Pair<String, String>> nameEntryList = new ArrayList<>();
+          for (JarEntry entry : listOfAllClasses)
+            nameEntryList.addAll(fetchAllMethodSignaturesForyaml(jarFile,entry));
+          ArrayList<Map<String,String>> entryResult = new ArrayList<>();
+
+          for(Pair<String, String> entry : nameEntryList)
+            entryResult.add(Map.ofEntries(entry("name",entry.first),entry("entryPoint",entry.second)));
+
+          Map<String, ArrayList<Map<String,String>>> dataMap = new HashMap<>();
+          dataMap.put("properties",entryResult);
+          final FileWriter writer = new FileWriter(args[2]+".yaml");
+          yaml.dump(dataMap, writer);
+          break;
+        }
         case "test": {
           TestArguments arguments = new TestArguments(args);
+          String entryPoint = null;
           // 1. Run Tests and obtain coverage
           RepoTool rt = maybeObtainTool(arguments);
-          List<Pair<String, String>> coverageFilesAndEntryPoints = rt.obtainCoverageFilesAndEntryPoints();
+          StaticCallgraph callgraph = deserializeStaticCallGraph(arguments);
+          List<Pair<String, ?>> coverageFilesAndEntryPointsShorthand = rt.obtainCoverageFilesAndEntryPoints();
+          List<Pair<String, String>> coverageFilesAndEntryPoints=new ArrayList<>();
+          for(Pair<String, ?> s : coverageFilesAndEntryPointsShorthand) {
+            Pair<String,String> result=new Pair<>(s.first,null);
+            if(s.second.getClass().toString().equals("class java.lang.String")){
+              entryPoint= (String) s.second;
+            }
+            else{
+              try {
+                Optional<String> returnType = Optional.empty();
```

Comment:
Add check for `s.second instanceof ArrayList`

Reply:
Done